batcher/batcher_test.go

```diff
@@ -0,0 +1,105 @@
+package batcher
+
+import (
+	"testing"
+	"time"
+
+	"github.com/influxdb/influxdb"
+)
+
+// TestBatch_Size ensures that a batcher generates a batch when the size threshold is reached.
+func TestBatch_Size(t *testing.T) {
+	batchSize := 5
+	batcher := NewBatcher(batchSize, time.Hour)
+	if batcher == nil {
+		t.Fatal("failed to create batcher for size test")
+	}
+
+	in := make(chan influxdb.Point)
+	out := make(chan []influxdb.Point)
+
+	go batcher.Start(in, out)
+
+	var p influxdb.Point
+	for i := 0; i < batchSize; i++ {
```

Comment:
You're right that it would run in a goroutine, but I don't fully follow what you mean by "really waits". Can you expand?